### PR TITLE
[WIP] Fixes getflaticon

### PR DESCRIPTION
**Pull request by silicons**
wip because i'm still finding the most performance efficient way of doing this.

### CODE REVIEW

File: code/__HELPERS/icons.dm [hunk 1]
```diff
@@ -706,8 +706,43 @@ world
 		((hi3 >= 65 ? hi3-55 : hi3-48)<<4) | (lo3 >= 65 ? lo3-55 : lo3-48),
 		((hi4 >= 65 ? hi4-55 : hi4-48)<<4) | (lo4 >= 65 ? lo4-55 : lo4-48))
 
+var/global/debug_g
```
Comment (Cyberboss) [inline code/__HELPERS/icons.dm]: kill urself

Comment (kevinz000) [reply code/__HELPERS/icons.dm]: its wip no bulli!

File: code/__HELPERS/icons.dm [hunk 2]
```diff
@@ -706,8 +706,43 @@ world
...
+	var/r = isnull(override)? rand(1, 100) : override
+	switch(r)
+		if(1 to 33)
+			getFlatIcon_1(A, B, C, D, E, F, G)
+		if(34 to 66)
+			getFlatIcon_2(A, B, C, D, E, F, G)
+		if(67 to 99)
+			getFlatIcon_3(A, B, C, D, E, F, G)
+		if(100)
+			getFlatIcon(A, B, C, D, E, F, G, rand(1, 100))
```
Comment (ExcessiveUseOfCobblestone) [inline code/__HELPERS/icons.dm]: ever considered rand(1,3) so you aren't throwing your values off?

Comment (kevinz000) [reply code/__HELPERS/icons.dm]: it's fiiiiiiiiiiiiiiiiiiiiiiiiiiiiiine

**Pull request closed without merging**